include/zephyr/drivers/gpio.h

```diff
@@ -555,6 +555,20 @@ __subsystem struct gpio_driver_api {
  * @endcond
  */
 
+/**
+ * @brief Validate that GPIO port is ready.
+ *
+ * @param spec GPIO specification from devicetree
+ *
+ * @retval true if the GPIO spec is ready for use.
+ * @retval false if the GPIO spec is not ready for use.
+ */
+static inline bool gpio_is_ready(const struct gpio_dt_spec *spec)
+{
+	/* Validate port is ready */
+	return device_is_ready(spec->port);
```

Comment:
Why not simply call `device_is_ready()`?

My one concern is that a GPIO isn't necessarily ready to use just because it has been probed. All of the various configuration bits need to be set as well.

Reply:
> Why not simply call `device_is_ready()`?
>

We have a similar function for SPI that verifies the device object in the same way which is convenient to use. Also if there would be a case when `gpio_dt_spec` struct will change the already existing `port` member to another one there would be only 1 place to change it instead of all of the existing drivers, tests and etc.

> My one concern is that a GPIO isn't necessarily ready to use just because it has been probed. All of the various configuration bits need to be set as well.

It's more about checking if the device object that was retrieved by "DEVICE_DT_GET" macro (which does not include the readiness checks) is in a valid state before the use.

See:
<pre>
#define GPIO_DT_SPEC_GET_BY_IDX(node_id, prop, idx)                            \
        {                                                                      \
                .port = DEVICE_DT_GET(DT_GPIO_CTLR_BY_IDX(node_id, prop, idx)),\
                .pin = DT_GPIO_PIN_BY_IDX(node_id, prop, idx),                 \
                .dt_flags = DT_GPIO_FLAGS_BY_IDX(node_id, prop, idx),          \
        }
</pre>